File: embrace/online-boutique/src/utils/money.js
/**
 * @param {{ currencyCode?: string, currency_code?: string, units?: number|string, nanos?: number }|undefined} m
 */
export function formatMoney(m) {
  if (!m) {
    return '—'
  }
  const code = m.currencyCode ?? m.currency_code ?? ''
  const units = typeof m.units === 'string' ? parseInt(m.units, 10) : (m.units ?? 0)
  const nanos = m.nanos ?? 0
  const frac = Math.abs(nanos) / 1e9
  const sign = units < 0 ? -1 : 1
  const absU = Math.abs(units)
  const total = absU + frac
  return `${code} ${sign < 0 ? '-' : ''}${total.toFixed(2)}`
}

File: embrace/online-boutique/flutter/lib/utils/money.dart
/// Same formatting as `src/utils/money.js`.
String formatMoney(Map<String, dynamic>? m) {
  if (m == null) return '—';
  final code = (m['currencyCode'] ?? m['currency_code'] ?? '').toString();
  final unitsRaw = m['units'];
  final units = unitsRaw is String ? int.tryParse(unitsRaw) ?? 0 : (unitsRaw as num?)?.toInt() ?? 0;
  final nanos = (m['nanos'] as num?)?.toInt() ?? 0;
  final frac = nanos.abs() / 1e9;
  final sign = units < 0 ? -1 : 1;
  final absU = units.abs();
  final total = absU + frac;
  final prefix = sign < 0 ? '-' : '';
  return '$code $prefix${total.toStringAsFixed(2)}';
}
